kv/range_cache.go

```diff
@@ -221,3 +230,38 @@ func (rmc *rangeDescriptorCache) getCachedRangeDescriptorLocked(key proto.Key) (
 	}
 	return metaEndKey, rd
 }
+
+// clearOverlappingCachedRangeDescriptors looks up and clears any
+// cache entries which overlap the specified key or descriptor.
+func (rmc *rangeDescriptorCache) clearOverlappingCachedRangeDescriptors(key, metaKey proto.Key, desc *proto.RangeDescriptor) {
+	if desc.StartKey.Equal(desc.EndKey) { // True for some unittests.
+		return
+	}
+	// Clear out any descriptors which subsume the key which we're going
+	// to cache. For example, if an existing KeyMin->KeyMax descriptor
+	// should be cleared out in favor of a KeyMin->"m" descriptor.
+	k, v, ok := rmc.rangeCache.Ceil(rangeCacheKey(metaKey))
+	if ok {
+		desc := v.(*proto.RangeDescriptor)
+		if log.V(1) {
+			log.Infof("considering overlapping descriptor: key=%s desc=%s", k, desc)
+		}
+		addrKey := keys.KeyAddress(key)
+		if !addrKey.Less(desc.StartKey) && !desc.EndKey.Less(addrKey) {
+			if log.V(1) {
+				log.Infof("clearing overlapping descriptor: key=%s desc=%s", k, desc)
+			}
+			rmc.rangeCache.Del(k.(rangeCacheKey))
+		}
```

Comment:
don't you think it's worth having a balancing log line to the "considering" log?